### ISSUE TITLE
Error on fresh project

### ISSUE THREAD

**Issue by jhorst11** [aspnet/JavaScriptServices#918]
Creating a fresh project and getting the following error in Development environment:

in [at-loader] ./ClientApp/boot-server.tsx:43:25 
    TS2345: Argument of type '{ html: string; globals: { initialReduxState: ApplicationState; }; }' is not assignable to parameter of type 'RenderToStringResult | RedirectResult | PromiseLike<RenderResult>'.
  Type '{ html: string; globals: { initialReduxState: ApplicationState; }; }' is not assignable to type 'PromiseLike<RenderResult>'.
    Property 'then' is missing in type '{ html: string; globals: { initialReduxState: ApplicationState; }; }'.

**Comment by Tonvengo** [comment]
Same issue here #917

**Comment by lee-alderdice** [comment]
Same issue but on VS2017.  Using dotnet run does successfully run the application however code changes are not reflected in the browser without total restarts of applications and so on.  This error started for me when I tried an upgrade to Angular 4.0 not knowing the problems that would subsequently ensue.

**Comment by abstract-christos** [comment]
same here with Angular 4 

Argument of type '(params: BootFuncParams) => Promise<{ html: string; globals: { [key: string]: any; styles: string...' is not assignable to parameter of type 'BootFunc'.
      Type 'Promise<{ html: string; globals: { [key: string]: any; styles: string; title: string; meta: strin...' is not assignable to type 'Promise<RenderToStringResult>'.
        Type '{ html: string; globals: { [key: string]: any; styles: string; title: string; meta: string; }; }' is not assignable to type 'RenderToStringResult'.
          Property 'statusCode' is missing in type '{ html: string; globals: { [key: string]: any; styles: string; title: string; meta: string; }; }'.

**Comment by fplaten** [comment]
We had the same issue and downgraded the npm package to 2.0.3. This fixed it for us.

**Comment by DanHarman** [comment]
Quick fix is to cast away the type. Not sure of underlying issue:

```
            params.domainTasks.then(() => {
                resolve({
                    html: renderToString(app),
                    globals: { initialReduxState: store.getState() }
                } as any);
            }, reject); // Also propagate any errors back into the host application
```

**Comment by lee-alderdice** [comment]
Hi @DanHarman.  Would you mind posting a fuller boot-server file.  I am only learning and finding it hard to update with these changes.  Thanks so much.

**Comment by DanHarman** [comment]
@lee-alderdice Hi, the only difference is adding `statusCode: 200` in at line 45 of boot-server assuming no other changes.

**Comment by MarkPieszak** [comment]
You can revert to aspnet-prerendering 2.0.3 to fix it for now as well.

**Comment by SteveSandersonMS** [comment]
Sorry about that. I've just published `aspnet-prerendering` 2.0.5 which fixes this.
